src/Symfony/Component/HttpKernel/HttpKernel.php

```diff
@@ -239,7 +240,18 @@ private function handleThrowable(\Throwable $e, Request $request, int $type): Re
                 $response->setStatusCode($e->getStatusCode());
                 $response->headers->add($e->getHeaders());
             } else {
-                $response->setStatusCode(500);
+                $class = new \ReflectionClass($e);
+                $attributes = $class->getAttributes(HttpException::class, \ReflectionAttribute::IS_INSTANCEOF);
+
+                if (0 === \count($attributes)) {
+                    $response->setStatusCode(500);
+                } else {
+                    /** @var HttpException $instance */
+                    $instance = $attributes[0]->newInstance();
+
+                    $response->setStatusCode($instance->statusCode);
+                    $response->headers->add($instance->headers);
+                }
```

Comment:
The logic to set the status code & headers should be placed in the ErrorListener (like it's currently done for similar features) so that the ErrorController can handle them properly.

Reply:
There is already logic for setting the status code and headers in the HttpKernel, just above the new code, where it's checking if the exception is implementing the HttpExceptionInterface. - https://github.com/symfony/symfony/pull/48352/files#diff-8a470a7186085672d64cd3b87c349715f4692dc39576b158aeec7a4a817e7a0bR239-R242 
This just extending that to be able to support the attributes.

As I saw, the ErrorListener is only wrapping the needed exception inside a HttpException, which is later used here to set the status code and the headers.

Reply:
I think (and correct me if I'm mistaken @yceruto), with the current implementation, using a standard Symfony app, the exceptions won't be rendered correctly. The logic in HttpKernel is sort of a failsafe.

Reply:
Just saw that the status code and headers are also being set in `\Symfony\Component\ErrorHandler\Exception\FlattenException::createFromThrowable`, which is used in the ErrorListener. I will add the logic there and check if it would still be needed in the HttpKernel as well.